### PR TITLE
ROX-20353: Reconcile image integrations

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

* Depends on: #8310 

Adding `Image Integrations` to the `reconciliationStore` and reconcile them upon connection with Central.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

 * [ ] CI
 * [ ] New unit tests added
 * [ ] Manual:
   * Deploy ACS
   * Create a secret that will trigger a `ImageIntegration` event: `kubectl create secret docker-registry test --docker-username=test --docker-password=1234 --docker-server=registry.test:5000`
   * Scale down Sensor
   * Delete the secret
   * Scale up Sensor
   * The secret should not appear in the `image_integrations` table

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: sensor/kubernetes/listener/resources/secrets.go [hunk 1]
```diff
@@ -284,6 +288,11 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 						ImageIntegration: ii,
 					},
 				})
+				if action == central.ResourceAction_REMOVE_RESOURCE {
```
Comment (fredrb) [inline sensor/kubernetes/listener/resources/secrets.go:291]: It looks odd that this event is generated both at the deployments and secrets handler.

**Review by fredrb — APPROVED** [review]
I see no blockers. Thanks for addressing this quickly!

File: sensor/kubernetes/listener/resources/secrets.go (hunk 1 above)
Comment (lvalerom) [reply sensor/kubernetes/listener/resources/secrets.go:291]: And `REMOVE` events are only handled if `ourcedAutogeneratedIntegrations` is enabled.

**Comment by vikin91** [comment]
@lvalerom can this be merged?

**Comment by lvalerom** [comment]
> @lvalerom can this be merged?

We're going with a different approach. I'll close the PR.

**Pull request closed without merging**